test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+
+module Test.Integration.Scenario.Wallets
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Cardano.Wallet.Primitive.Types
+    ( WalletBalance (..), WalletDelegation (..), WalletState (..) )
+import Data.Quantity
+    ( Quantity (..), zeroPercentage )
+import qualified Network.HTTP.Types.Status as HTTP
+import Test.Hspec
+    ( SpecWith, it )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , addressPoolGap
+    , balance
+    , balanceAvailable
+    , balanceTotal
+    , delegation
+    , expectFieldEqual
+    , expectResponseCode
+    , json
+    , request
+    , state
+    , walletName
+    )
+
+spec :: SpecWith Context
+spec = do
+    it "Create a wallet" $ \ctx -> do
+
+        let payload = Json [json| {
+                "name": "1st Wallet",
+                "mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
+                "mnemonic_second_factor": ["attract", "tornado", "slender", "pumpkin", "clown", "announce", "term", "winner", "ready"],
+                "passphrase": "Secure Passphrase",
+                "address_pool_gap": 20
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        print (show r)
+        expectResponseCode @IO HTTP.status200 r
+        expectFieldEqual walletName "1st Wallet" r
+        expectFieldEqual addressPoolGap 20 r
+        expectFieldEqual balance (WalletBalance
+                                    { available = Quantity 0
+                                    , total = Quantity 0}) r
+        expectFieldEqual balanceAvailable (Quantity 0) r
+        expectFieldEqual balanceTotal (Quantity 0) r
+        expectFieldEqual delegation (NotDelegating) r
+        expectFieldEqual state (Restoring (Quantity zeroPercentage)) r
+        -- expectFieldEqual passphraseLastUpdate "2019-04-10 11:48:31.652664497 UTC" r
+        -- expectFieldEqual walletId 4bba596017a63e52a4cf6caaadf8a670bfe4745b r
```

Comment:
I have defined lenses for those two, but don't know how to use them at this point :).
Perhaps walletId does not need to be checked here as there are other tests for that, not sure also about the usefulness of the `passphraseLastUpdate`